### ISSUE TITLE
libGPML- library to read and write GPML files

### ISSUE THREAD

**Issue by mkutmon** [nrnb/GoogleSummerOfCode#66]
### Background

GPML (Graphical Pathway Markup Language) is the XML format used by WikiPathways and PathVisio to store graphical and semantic pathway data. [WikiPathways](http://www.wikipathways.org) is a collaborative pathway database that is build on Mediawiki and [PathVisio](http://www.pathvisio.org) is a standalone pathway editor implemented in Java. Currently the GPML reading and writing functionality is integrated in the PathVisio core module. This module also contains lots of PathVisio specific aspects (view, settings, properties,...). 

### Goal

The goal of this project is to create a separate GPML library to read and write GPML files in Java which can be used by PathVisio, the WikiPathways app for Cytoscape and many other applications. Currently it is required to include the whole PathVisio core module which is much bigger and more specific than a general GPML library should be so it can be easily integrated in other applications. 
Ideally, libGPML also supports the export (and maybe import) of common graph and network formats - e.g. export to XGMML or SIF. 

### Difficulty Level: 1

No prior experience with PathVisio is required, though students should familiarize themselves with the basics of available pathway information and the PathVisio code base early on during the application process.

### Potential Mentors

[Martina Kutmon](mailto:mkutmon@gmail.com)

### Contact

[Martina Kutmon](mailto:mkutmon@gmail.com)

**Comment by Vetox** [comment]
What would be the best way to move forward with this issue ?
I'm interested in contributing to this project and solving this issue seems the most suited to my current skill set.
I've contacted you through email. Is there another more convenient way to contact you ?

**Comment by shreyanshk** [comment]
Hi,
Let me introduce myself to you.
I’m Shreyansh Khajanchi, undergraduate student of class of 2019 at VIT University, Chennai Campus, India. I’m an open source lover and Linux enthusiast. I’m a self-taught programmer with Java, PHP, Python, C, C++, C# including scripting languages such as Bash, JavaScript and Microsoft's Powershell up my sleeves. I'm also well versed with Linux and I have used myriad of distributions in past including but not limited to Mint, Fedora, Debian, Ubuntu etc and I'm personally using Arch Linux right now as my preferred distribution. This is just a tip of the iceberg and I can barely contain myself right now.

With that out of way, here’s the important stuff.

I’m applying for GSoC’17. For starters like me, I have a feeling that this is a nice idea to get indulged.
What is seek from you is the knowledge about how should I proceed with the applying process. Which include questions like:
1)	What are you expecting in the proposal document which I’ll submit on GSoC website?
2)	Will you be willing to review it over the email before I submit it on GSoC website?
3)	What are the things obvious to the experienced people which are not so obvious to the newcomers?
4)	I believe having a well understood guideline of what is expected of each other beforehand is extremely important for smooth functioning. What exactly can I expect from you and GSoC in general?

My first step to becoming a part of this is nailing the applying process, therefore *actually* getting selected is important. Hence, to fulfil that, I’ve written this post and come to you.
Hoping to hear soon from you. :)
